modules/upload/upload.service.js

```diff
@@ -35,83 +63,40 @@ class UploadService {
-                        }
-                    })
-                })
-                image.resize(1080, Jimp.AUTO).getBuffer(image.getMIME(), (err, buffer) => {
-                    const
-                        blobName = `medium_${id}_${filename}`,
-                        stream = getStream(buffer),
-                        streamLength = buffer.length
-
-                    blobService.createBlockBlobFromStream(containerName, blobName, stream, streamLength, err => {
-                        if (err) {
-                            throw new Error(err)
-                        }
-                    })
-                })
+            this.uploadResizedImage(1920, createName('large'), image)
 
-                image.resize(768, Jimp.AUTO).getBuffer(image.getMIME(), (err, buffer) => {
-                    const
-                        blobName = `small_${id}_${filename}`,
-                        stream = getStream(buffer),
-                        streamLength = buffer.length
-
-                    blobService.createBlockBlobFromStream(containerName, blobName, stream, streamLength, err => {
-                        if (err) {
-                            throw new Error(err)
-                        }
-                    })
-                })
+            this.uploadResizedImage(1080, createName('medium'), image)
 
-                image.resize(128, Jimp.AUTO).getBuffer(image.getMIME(), (err, buffer) => {
-                    const
-                        blobName = `thumbnail_${id}_${filename}`,
-                        stream = getStream(buffer),
-                        streamLength = buffer.length
-
-                    blobService.createBlockBlobFromStream(containerName, blobName, stream, streamLength, err => {
-                        if (err) {
-                            throw new Error(err)
-                        }
-                    })
-                })
-            })
+            this.uploadResizedImage(768, createName('small'), image)
+
+            this.uploadResizedImage(128, createName('thumbnail'), image)
 
             return {
                 prefixUrl: process.env.IMAGE_LINK,
                 fileNames: {
-                    large: `large_${id}_${filename}`,
-                    medium: `medium_${id}_${filename}`,
-                    small: `small_${id}_${filename}`,
-                    thumbnail: `thumbnail_${id}_${filename}`,
+                    large: createName('large'),
+                    medium: createName('medium'),
+                    small: createName('small'),
+                    thumbnail: createName('thumbnail'),
                 }
             }
         })
     }
 
     async deleteFiles(files) {
-        return files.map(fileName => {
+        return files.map(async fileName => 
+            await new Promise((resolve, reject) =>
                 blobService.deleteBlobIfExists(containerName, fileName, (err, res) => {
                     if(!err){
```

Comment:
If you firstly call `reject` in condition `if (err) {...}` then do it here first too, please

Reply:
fixed